### PR TITLE
[crew] Introduce initial crew library

### PR DESCRIPTION
**Pull request by seanshpark**
This will introduce initial crew library that will provide
config ini/json file import/export functionalities.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #6230, from draft #6231

File: compiler/crew/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# crew
+
+_crew_ is circle partitioning Configuration REader and Writer library.
```
Comment (mhs4670go) [inline compiler/crew/README.md:3]:
```suggestion
_crew_ is circle partitioning Configuration Reader and Writer library.
```

Comment (seanshpark) [reply compiler/crew/README.md:3]: `RE` is intentionally set capital for `cREw`

**Comment by seanshpark** [comment]
I need to sync with draft: there are some changes...

File: compiler/crew/src/ConfigIni.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+{
+  assert(data != nullptr);
+  assert(length > 0);
+
+  auto buffer = std::vector<char>();
+  buffer.reserve(length + 1);
+  char *pbuffer = buffer.data();
+  memcpy(pbuffer, data, length);
+  // add null at end to be sure
+  *(pbuffer + length) = 0;
```
Comment (mhs4670go) [inline compiler/crew/src/ConfigIni.cpp:39]:
```suggestion
  std::vector<char> buffer(data, data + length);
  buffer.push_back('\0');
  char *pbuffer = buffer.data();
```

Comment (seanshpark) [reply compiler/crew/src/ConfigIni.cpp:39]: This doesn compile: `error: no matching function for call to ‘std::vector<char>::vector(const char*&, size_t&)’`

**Comment by seanshpark** [comment]
I've renamed `ConfigIni` to `PConfigIni` as it's for `Partition Config`

File: compiler/crew/src/ConfigIni.cpp (hunk 2 above)
Comment (mhs4670go) [reply compiler/crew/src/ConfigIni.cpp]: @seanshpark Hmmm.. that's weird. It can be compiled in my local pc.

**Review by mhs4670go — APPROVED** [review]
LGTM

File: compiler/crew/src/ConfigIni.cpp (hunk 2 above)
Comment (seanshpark) [reply compiler/crew/src/ConfigIni.cpp]: Maybe I have some other problems. If you like, please post a revision after this lands...

**Pull request merged**